### DISCUSSION TITLE
slicing by distance

### DISCUSSION THREAD

**Discussion by hanahg** [MDAnalysis/mdanalysis#4364] [Q&A]
Hi,

I would like to slicing a trajectory by distance, is it possible?

**Comment by IAlibay** [comment]
@hanahg this seems to be more of a usability question so I will convert this to a discussion item rather than an issue.

**Comment by IAlibay** [comment]
@hanahg Could you offer more details about what you are trying to do? Specifically, what do you mean by "distance"?

**Comment by hanahg** [comment]
In fact, I should calculate and draw a curve of electrostatic charge of a Nan-oparticle(NP). For that I should calculate the charge of NP from the center, then calculate the electrostatic charges(as it reported in a paper). I don't know how but maybe by calculating charge of each slice as a group.

**Reply by orbeckst** [reply]
If you can formulate your problem in terms of equations then we can help you and tell you how you can implement the equations with MDAnalysis.

If you only have a vague idea that are expressed in words then we might just not have any idea how to help and then you might not get any replies. The more precise and focused your question is, the more likely it's generally that someone can help.

**Reply by hanahg** [reply]
I have explained "I should calculate and draw a curve of electrostatic charge of a Nan-oparticle(NP). For that I should calculate the charge of NP from the center, then calculate the electrostatic charges". 
please tell me how to calculate distance of atoms from the center of Nanoparticle, I mean position (0 0 0)

**Reply by RMeli** [reply]
> I have explained

@hanahg if someone asks for additional explanation, it is likely because they did not understand what you want to do. It is unclear to me what you are trying to achieve from your explanation.

> please tell me how to calculate distance of atoms from the center of Nanoparticle, I mean position (0 0 0)

What do you mean by position `(0, 0, 0)`? If your nanoparticle is centred at the origin, then the distance of atoms from the center, will be the distance of the atoms from the origin. If you have an atom group `ag`, you can get the position of all the atoms as a NumPy array with `ag.positions`.

If the nanoparticle is not actually centred at the origin, you can select the nanoparticle using `select_atoms` and compute its center of mass with `center_of_mass()` or its center of geometry with `center_of_geometry()`.

See [User Guide - AtomGroup positions and methods](https://userguide.mdanalysis.org/stable/examples/quickstart.html#AtomGroup-positions-and-methods) for some more details.
